src/editor/VideoViewer.js

```diff
@@ -0,0 +1,324 @@
+        var fullUrl = UrlCache.getFilename(UrlCache.getUrl(file.fullPath));
+        var root = StartupState.project("root");
+        return fullUrl.replace(root,"").replace("/","");
+    }
+
+    /**
+     * Check if it's a video file TODO: refactor
+     */
+    function isVideo(fullPath) {
+        return Content.isVideo(Path.extname(fullPath));
+    }
+
+    /**
+     * VideoView objects are constructed when a video is opened
+     * @see {@link Pane} for more information about where VideoViews are rendered
+     *
+     * @constructor
+     * @param {!File} file - The video file object to render
+     * @param {!jQuery} container - The container to render the video view in
+     */
+    function VideoView(file, $container) {
+        var that = this;
+
+        this.file = file;
+        this.container = $container;
+
+        // Set defaults, and keep track of which video element attributes are enabled.
+        this.videoTagSettings = {
+            controls: true,
+            autoplay: false,
+            loop: false,
+            muted: false
+        }
+
+        // Gets the video type for sample markup
+        this.videoType = Path.extname(this.file.fullPath).replace(".","");
+
+        this.$el = $(Mustache.render(VideoViewTemplate, {
+            videoUrl: _getVideoUrl(file),
+            videoType: this.videoType,
+            Strings: Strings
+        }));
+
+        $container.append(this.$el);
+
+        this._naturalWidth = 0;
+        this._naturalHeight = 0;
+
+        this.relPath = ProjectManager.makeProjectRelativeIfPossible(this.file.fullPath);
+
+        this.$videoEl = this.$el.find("video");
+        this.$videoWrapperEl = this.$el.find(".video-wrapper");
+        this.$videoMarkupEl = this.$el.find("pre");
+        this.$videoData = this.$el.find(".video-data");
+
+        this.$videoEl.attr("src", _getVideoUrl(this.file));
+        this.$videoEl.one("canplay", _.bind(this._onVideoLoaded, this));
+        this.$videoEl.on("error", _.bind(console.error, console));
+
+        this.$el.find(".tag-options").on("change", "input", function(){
+            var attribute = $(this).attr("setting");
+            var isEnabled = $(this).is(":checked");
+            that.videoTagSettings[attribute] = isEnabled;
+            that.updateVideoTagMarkup(true);
+        });
+
+
+        this.updateVideoTagMarkup();
+
+        _viewers[file.fullPath] = this;
+    }
+
+    // Updates the markup used in the preview & the sample markup below
+    VideoView.prototype.updateVideoTagMarkup = function(reload){
+
+        var videoTagAttributesString = "";
+
+        for(var k in this.videoTagSettings) {
+            if(this.videoTagSettings[k]) {
+                videoTagAttributesString = videoTagAttributesString + " " + k;
+                this.$videoEl.attr(k,"");
+            } else {
+                this.$videoEl.removeAttr(k);
+            }
+        }
+
+        var videoTagMarkup =
+`<video`+ videoTagAttributesString +`>
```

Comment:
We don't use template strings anywhere in Brackets, so probably better to use a regular string literal.